src/cpp/rtps/builtin/discovery/participant/timedevent/ResendParticipantProxyDataPeriod.cpp

```diff
@@ -27,52 +27,73 @@
 
 
 namespace eprosima {
-namespace fastrtps{
+namespace fastrtps {
 namespace rtps {
 
 
-ResendParticipantProxyDataPeriod::ResendParticipantProxyDataPeriod(PDPSimple* p_SPDP,
-        double interval):
-    TimedEvent(p_SPDP->getRTPSParticipant()->getEventResource().getIOService(),
-            p_SPDP->getRTPSParticipant()->getEventResource().getThread(), interval),
-    mp_PDP(p_SPDP)
+ResendParticipantProxyDataPeriod::ResendParticipantProxyDataPeriod(
+        PDPSimple* p_SPDP,
+        const BuiltinAttributes& config)
+    : TimedEvent(
+            p_SPDP->getRTPSParticipant()->getEventResource().getIOService(),
+            p_SPDP->getRTPSParticipant()->getEventResource().getThread(),
+            0)
+    , pdp_(p_SPDP)
+    , standard_period_(config.leaseDuration_announcementperiod)
+    , initial_announcements_(config.initial_announcements)
+{
+    if ((initial_announcements_.count > 0) && (initial_announcements_.period <= c_TimeZero))
```

Comment:
If the user passes 0 or negative we setup 1ms, maybe we should warn about this.
Note that as tinyxml2 works passing a negative count is actually passing a large positive one.

Reply:
> maybe we should warn about this

done in commit 43ff2e7254a924debf45ea42d12056a052ca265a